nvtabular/ops/categorify.py

```diff
@@ -196,28 +199,72 @@ def __init__(
         self.stat_name = "categories"
         self.encode_type = encode_type
         self.search_sorted = search_sorted
+        self.hash_type = hash_type
 
         if self.search_sorted and self.freq_threshold:
             raise ValueError(
                 "cannot use search_sorted=True with anything else than the default freq_threshold"
             )
+        if num_buckets:
+            if isinstance(num_buckets, dict):
```

Comment:
Lets not have two levels of if statements here - can we get rid of the top level `if num_buckets` statement in favour of the checks below ?